docs/auth.md

```diff
@@ -0,0 +1,73 @@
+# Authorization Guide
+
+## Overview
+
+Optimizely Agent supports authorization workflows based on OAuth and JWT standards, allowing you to protect access to its API and Admin interfaces.
+
+There are three modes of operation:
+
+### Issuer & Validator
+Access tokens are issued by Agent itself, using a [Client Credentials grant](https://www.oauth.com/oauth2-servers/access-tokens/client-credentials/). Access tokens are signed and validated using the HS256 algorithm with a signing secret provided in configuration.
+
+### Validator-only
+Agent validates access tokens that were issued elsewhere. Access tokens are validated with public keys fetched from a [JWKS](https://tools.ietf.org/html/rfc7517) URL provided in configuration.
+
+### No authorization (default)
+The interface is publicly available.
+
+## Configuration
+- The API and Admin interfaces are each independently configured to run in one of the above-mentioned modes of operation.
+- Authorization configuration is located under the `auth` key
+- Each mode of operation has its own set of configuration properties, described below.
+
+### Issuer & Validator
+The configuration properties pertaining to Issuer & Validator mode are listed below:
+
+|Property Name|Description|
+|---|---|
+|ttl|Time-to-live of access tokens issued|
+|hmacSecrets|Secret used to sign issued access tokens, using the HMAC SHA256 algorithm|
+|clients|Array of client credentials, any of which can be exchanged for an access token. Each object in the array should have `"id"` and `"secret"` string properties.|
+
+### Validator-only
+The configuration properties pertaining to Validator-only mode are listed below:
+
+|Property Name|Description|
+|---|---|
+|jwksURL|URL from which public keys should be fetched for token validation|
+
+### No authorization (default)
+The API & Admin interfaces run with no authorization when no `auth` configuration is given.
+
+### Example configuration file (yaml)
```

Comment:
It seems like above there are two separate options, but here there is only one yaml file showing both examples. It might be more clear if we have two separate yaml examples for each option if I understand this correctly.

Reply:
Yep, that's right - I'll split it into two example yaml files.